service/tablet_allocator.cc

```diff
@@ -264,6 +264,75 @@ class load_balancer {
         }
     };
 
+    // Maximum tablet size works as the split threshold, whereas minimum size works as the
+    // merge threshold. To prevent oscillations between split and merge, hysteresis is applied
+    // to the decision.
+    // TODO: possibly make target_max_tablet_size a config to help writing tests that stresses resize path.
+    static const size_t target_max_tablet_size = 10UL * 1024 * 1024 * 1024;
+    static const size_t target_min_tablet_size = double(target_max_tablet_size / 2) * 0.6;
+
+    // This limits defines the maximum number of resize requests happening at a point in time.
+    // Due to power-of-two constraint, resize is always a cluster-wide operation meaning that
+    // a table will have all of its tablets resized, not only a subset. Therefore, it's important
+    // to set a limit on the operation, to avoid interfering with the foreground activity.
+    static const size_t max_concurrent_resize_requests = 1;
+
+    struct cluster_resize_stats {
+        using table_and_avg_size = std::pair<table_id, size_t>;
+        std::vector<table_and_avg_size> tables_need_resize;
+        std::vector<table_and_avg_size> tables_being_resized;
+
+        bool table_needs_resize(size_t avg_tablet_size) const {
+            // FIXME: ignore merge request if tablet_count == initial_tablets.
```

Comment:
Idea: ignore if tablet_count < initial_tablets, and then remove initial_tablets. Then the hint disappears once we reach the target, then the algorithm takes over.

Reply:
So we are only looking at the average? Consider target is 5GB per tablet, we have 4 tablets of size [0, 0, 0, 19GB] , the average is still less than 5GB, but we'd better split in this case.

Reply:
Yes. In the case you show, all the data is likely in one huge partition, so splitting doesn't help.

With power-of-two we have very little control. If/when we remove the limitation we can consider looking at tablet sizes. I don't think it's very helpful, since tablet sizes fluctuate with compaction.